shared/types.go

```diff
@@ -136,6 +136,16 @@ type Micro struct {
 	Dev          string   `yaml:"dev,omitempty"`
 }
 
+type DiscoveryFrontmatter struct {
```

Comment:
I dont think the `DiscoveryFrontmatter` is right, since the struct also contains the content of the file.

Reply:
Maybe something like this would make more sense ?

```go
type DiscoveryFile struct {
    Content string
    Frontmatter DiscoveryFrontmatter
} 
```

Reply:
In the backend and in the API responses the content is just part of the same struct as the frontmatter though, as we don't care about how it was stored/separated in the file after parsing. It shouldn't be any different here.

I agree though that the name is not good, I'll rename it to `DiscoveryData`.